packages/ipfs-core/src/components/pin/remote.js

```diff
@@ -0,0 +1,441 @@
+
+  /**
+   * Remove a single pin from a remote pinning service.
+   * Fails if multiple pins match the specified criteria. Use rmAll to remove all pins that match.
+   *
+   * @param {object} options
+   * @param {string} options.service - name of a configured remote pinning service
+   * @param {Array<string>} options.cid - CID(s) to remove from remote pinning service
+   * @param {?Array<PinStatus>} options.status - only remove pins that have one of the specified statuses (queued, pinning, pinned, failed). Default: pinned
+   * @param {?number} options.timeout - request timeout (seconds)
+   * @returns {Promise<void>}
+   */
+  async rm (options) {
+    const { service, ...rmOpts } = options
+    if (!service) {
+      throw new Error('service name must be passed')
+    }
+    const svc = this.service.serviceNamed(service)
+    return svc.rm(rmOpts)
+  }
+
+  /**
+   * Remove all pins that match the given criteria from a remote pinning service.
+   *
+   * @param {object} options
+   * @param {string} options.service - name of a configured remote pinning service
+   * @param {Array<string>} options.cid - CID(s) to remove from remote pinning service
+   * @param {?Array<PinStatus>} options.status - only remove pins that have one of the specified statuses (queued, pinning, pinned, failed). Default: pinned
+   * @param {?number} options.timeout - request timeout (seconds)
+   * @returns {Promise<void>}
+   */
+  async rmAll (options) {
+    const { service, ...rmOpts } = options
+    if (!service) {
+      throw new Error('service name must be passed')
+    }
+    const svc = this.service.serviceNamed(service)
+    return svc.rmAll(rmOpts)
+  }
+}
+
+/**
+ * RemotePinServiceAPI provides methods to add, remove, and list the configured
+ * remote pinning services that are used by the remote pinning api.
+ */
+class PinRemoteServiceAPI {
+  constructor ({ config, swarm, peerId }) {
+    this.config = config
+    this.swarm = swarm
+    this.peerId = peerId
+
+    // TODO: read service config from IPFS config to construct remote service at init
+    this._services = {}
+  }
+
+  /**
+   * Adds a new remote pinning service to the set of configured services.
+   *
+   * @param {string} name - the name of the pinning service. Used to identify the service in future remote pinning API calls.
+   * @param {Object} options
+   * @param {string|URL} options.endpoint - the remote API endpoint URL
+   * @param {string} options.key - an API key that authorizes use of the remote pinning service
+   */
+  async add (name, options) {
+    if (this._services[name]) {
+      throw new Error('service already present: ' + name)
+    }
+
+    if (!options.endpoint) {
+      throw new Error('option "endpoint" is required')
+    }
+
+    if (!options.key) {
+      throw new Error('option "key" is required')
+    }
+
+    const svcOpts = Object.assign({ swarm: this.swarm, peerId: this.peerId }, options)
+    this._services[name] = new RemotePinningService(name, svcOpts)
```

Comment:
We would want to persist these (I think in) config so registered services are available across sessions.